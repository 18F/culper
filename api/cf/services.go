package cf

import (
	"encoding/json"
	"os"
	"strings"
)

// vcapService represents the JSON structure found in the CloudFoundry
// VCAP_SERVICES environment variable
type vcapService struct {
	UserProvided []service `json:"user-provided"`
}

// service represents the JSON structure which is common between the
// user provided services as well as other services
type service struct {
	Name        string            `json:"name"`
	Label       string            `json:"label"`
	Credentials map[string]string `json:"credentials"`
}

// UserService will attempt to pull credential information for a service by
// service name and credential name.
//
// If there is a VCAP_SERVICES environment variable it will attempt to parse
// its JSON value.
//
// Otherwise it will look for an environment variable in a normalized name
// and use its value:
//
//  - app-smtp => APP_SMTP
//  - app-postgres => APP_POSTGRES
func UserService(name, credential string) string {
	// If there is not a VCAP_SERVICES environment variable we check
	// if there is a one by itself.
	vcap := os.Getenv("VCAP_SERVICES")
	if vcap == "" {
		return os.Getenv(slugify(name, credential))
	}

	// The VCAP_SERVICES stores a JSON value which we can parse
	var v vcapService
	err := json.Unmarshal([]byte(vcap), &v)
	if err != nil {
		return ""
	}

	// Look for the user provided service and it's specific credential
	for _, s := range v.UserProvided {
		if s.Name == name {
			return s.Credentials[credential]
		}
	}

	return ""
}

// slugify provides a concatenation of strings to then perform
// a dirty normalization.
func slugify(slugs ...string) string {
	s := strings.Join(slugs, "_")
	s = strings.Replace(s, "-", "_", -1)
	return strings.ToUpper(s)
}
