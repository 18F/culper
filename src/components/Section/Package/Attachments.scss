@import 'eqip-colors';

.attachments {
  .file-upload-label {
    display: none;
    border: solid 3px $eapp-blue;
    border-radius: 5px;
    color: $eapp-blue;
    padding: 1rem;
    display: inline-block;
    width: fit-content;

    i {
      margin-left: 1rem;
    }
  }

  button {
    padding: 1.5rem 3rem;
  }

  table > tbody > tr > td {
    max-width: 64rem;
    width: 100%;
    border-top: 2px solid $eapp-grey;
    border-bottom: 2px solid $eapp-grey;
    border-left: none;
    border-right: none;

    button {
      background: #fff;
      color: $eapp-grey-darker;
      text-decoration: none;
      padding: 0;
      margin: 0;

      i {
        display: block;
        margin-bottom: 1rem;
      }
    }
  }
}
