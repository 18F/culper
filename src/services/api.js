import axios from 'axios'
import env from '../config/environment'

class Api {
  constructor () {
    this.proxy = axios.create({
      baseURL: env ? env.ApiBaseURL() : '/api',
      timeout: 5000
    })
  }

  /**
   * Helper method to extract query parameters from the url
   */
  getQueryValue (key) {
    let query = window.location.search.substring(1)
    let vars = query.split('&')
    let values = []

    for (let i = 0; i < vars.length; i++) {
      let pair = vars[i].split('=')
      if (pair[0] === key) {
        values.push(pair[1])
      }
    }

    if (values.length === 0) {
      return null
    } else if (values.length === 1) {
      return values[0]
    }

    return values
  }

  getCookieValue (key) {
    let cookies = document.cookie
    let vars = cookies.split(';')
    let values = []

    for (let i = 0; i < vars.length; i++) {
      let pair = vars[i].split('=')
      if (pair[0].trim() === key) {
        values.push(pair[1].trim())
      }
    }

    if (values.length === 0) {
      return null
    } else if (values.length === 1) {
      return values[0]
    }

    return values
  }

  getToken () {
    // Look for token in local storage
    let token = null
    if (this.supportForLocalStorage()) {
      token = window.localStorage.getItem('token')
    }

    // Look for token as cookie
    if (token === null) {
      token = this.getCookieValue('token')
    }

    // Look for token in query string
    if (token === null) {
      token = this.getQueryValue('token')
    }

    if (token === null && env && env.IsTest()) {
      token = window.token
    }

    return token
  }

  setToken (token) {
    if (this.supportForLocalStorage()) {
      window.localStorage.setItem('token', token)
    } else {
      document.cookie = 'token=' + token
    }
  }

  bearerToken () {
    return { 'Authorization': `Bearer ${this.getToken()}` }
  }

  supportForLocalStorage () {
    try {
      return 'localStorage' in window && window['localStorage'] !== null
    } catch (e) {
      return false
    }
  }

  information () {
    return this.proxy.get('/')
  }

  get (endpoint, secure = true) {
    const headers = secure ? { headers: this.bearerToken() } : {}
    return this.proxy.get(endpoint, headers)
  }

  post (endpoint, params = {}, secure = true) {
    const headers = secure ? { headers: this.bearerToken() } : {}
    return this.proxy.post(endpoint, params, headers)
  }

  saml () {
    return this.get(env.EndpointSaml(), false)
  }

  twoFactor (account, token) {
    if (token) {
      return this.post(env.EndpointTwoFactorVerify(account), { token: token })
    }

    return this.get(env.EndpointTwoFactor(account))
  }

  twoFactorReset (account) {
    return this.get(env.EndpointTwoFactorReset(account))
  }

  login (username, password) {
    return this.post(env.EndpointBasicAuthentication(), { username: username, password: password }, false)
  }

  refresh () {
    return this.post(env.EndpointRefresh())
  }

  save (payload) {
    return this.post(env.EndpointSave(), payload)
  }

  section (type) {
    return this.get(env.EndpointSection(type))
  }

  form () {
    return this.get(env.EndpointForm())
  }

  hash () {
    return this.get(env.EndpointFormHash())
  }

  validate (payload) {
    return this.post(env.EndpointValidate(), payload)
  }
}

const api = new Api()
export { api }
