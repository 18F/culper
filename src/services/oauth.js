class OAuth {
  constructor (service) {
    this.url = 'http://localhost:3000/auth/' + service
    this.token = ''
    this.expiration = null
  }

  getQueryValue (key) {
    let query = window.location.search.substring(1)
    let vars = query.split('&')
    let values = []

    for (let i = 0; i < vars.length; i++) {
      let pair = vars[i].split('=')
      if (pair[0] === key) {
        values.push(pair[1])
      }
    }

    if (values.length === 0) {
      return null
    } else if (values.length === 1) {
      return values[0]
    }

    return values
  }

  logout () {
    this.token = ''
    this.expiration = null
  }

  refreshToken () {
    if (this.token) {
            // TODO: Just request a refreshed token
      return
    }

    // TODO: Request authorization from OAuth provider
  }

  authenticated () {
    if (!this.token) {
      return false
    }

    let expired = !this.expiration || this.expiration <= new Date()
    if (expired) {
      return false
    }

    return true
  }
}

const GithubOAuth = new OAuth('github')
export { GithubOAuth, OAuth }
