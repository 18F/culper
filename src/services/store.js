import thunk from 'redux-thunk'
import rootReducer from '../reducers'
import { createStore, applyMiddleware } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import { historyMiddleware } from '../middleware/history'

const middleware = [thunk, historyMiddleware]

// Creates a redux store that defines the state tree for the application.
// See rootReducer for all sub-states.
let store
switch (process.env.NODE_ENV) {
case 'test':
  store = createStore(rootReducer)
  break
case 'production':
  store = createStore(rootReducer, applyMiddleware(...middleware))
  break
default:
  store = createStore(rootReducer, composeWithDevTools(applyMiddleware(...middleware)))
}

export default store
