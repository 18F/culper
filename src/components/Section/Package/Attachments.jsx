import React from 'react'
import FileSaver from 'file-saver'
import { connect } from 'react-redux'
import { i18n, env } from '../../../config'
import { api } from '../../../services'
import SectionElement from '../SectionElement'
import { Show, Field, Text, RadioGroup, Radio, Svg } from '../../Form'

export default class Attachments extends SectionElement {
  constructor (props) {
    super(props)
    this.state = {
      attachments: [],
      errorMessage: this.props.errorMessage
    }

    this.supportsBlobs = false
    try {
      this.supportsBlobs = !!new window.Blob
    } catch (e) {
      this.supportsBlobs = false
    }

    this.update = this.update.bind(this)
    this.updateAttachmentType = this.updateAttachmentType.bind(this)
    this.updateUploadDescription = this.updateUploadDescription.bind(this)
    this.updateOtherMethod = this.updateOtherMethod.bind(this)
    this.uploadFile = this.uploadFile.bind(this)
    this.delete = this.delete.bind(this)
    this.getStoredAttachments = this.getStoredAttachments.bind(this)
    this.displayAttachments = this.displayAttachments.bind(this)
  }

  componentDidMount () {
    this.getStoredAttachments()
  }

  update (queue) {
    this.props.onUpdate({
      AttachmentType: this.props.AttachmentType,
      UploadDescription: this.props.UploadDescription,
      OtherMethod: this.props.OtherMethod,
      ...queue
    })
  }

  updateAttachmentType (values) {
    this.update({ AttachmentType: values })
  }

  updateUploadDescription (values) {
    this.update({ UploadDescription: values })
  }

  updateOtherMethod (values) {
    this.update({ OtherMethod: values })
  }

  uploadFile (event) {
    const set = (attachment) => {
      // reset the file input
      this.refs.form.reset()

      // add the attachment
      const attachments = this.state.attachments
      attachments.push(attachment)
      this.setState({ attachments: attachments, errorMessage: '' }, () => {
        // clear out description value
        this.updateUploadDescription({ value: '' })
      })
    }

    const description = (this.props.UploadDescription || {}).value
    const file = this.refs.file.files[0]
    let formData = new window.FormData()
    formData.append('file', file)

    api.saveAttachment(formData).then((response) => {
      const attachmentID = response.data || 0
      if (attachmentID === 0) {
        return
      }
      if (!description) {
        set({
          id: attachmentID,
          filename: file.name,
          size: file.size,
          description: description
        })
        return
      }
      api.updateAttachment(attachmentID, description).then(response => {
        set({
          id: attachmentID,
          filename: file.name,
          size: file.size,
          description: description
        })
      }).catch(() => {
        this.setState({ errorMessage: i18n.t('application.attachments.upload.error.update') })
      })
    }).catch(() => {
      this.setState({ errorMessage: i18n.t('application.attachments.upload.error.save') })
    })
    event.preventDefault()
  }

  download (id) {
    if (!this.supportsBlobs) {
      return
    }

    const attachment = this.state.attachments.find(x => x.id === id)
    api.getAttachment(id).then((response) => {
      const blob = blobFromBase64(response.data, 'application/octet-stream')
      FileSaver.saveAs(blob, attachment.filename)
    }).catch(() => {
      this.setState({ errorMessage: i18n.t('application.attachments.upload.error.download') })
    })
  }

  delete (id) {
    api.deleteAttachment(id).then((response) => {
      let attachments = this.state.attachments
      attachments = attachments.filter(x => x.id !== id)
      this.setState({ attachments: attachments, errorMessage: '' })
    }).catch(() => {
      this.setState({ errorMessage: i18n.t('application.attachments.upload.error.delete') })
    })
  }

  getStoredAttachments () {
    api.listAttachments().then((response) => {
      this.setState({ attachments: response.data || [], errorMessage: '' })
    })
  }

  displayAttachments (items) {
    return items.map((x, i) => {
      return (
        <tr key={`attachment-${x.id}`}>
          <td>
            <Show when={this.supportsBlobs}>
              <a href="javascript:;;" aria-label={`Download ${x.filename}`} onClick={this.download.bind(this, x.id)}>
                <strong>{`${i+1}. `}{x.description ? `${x.description} - ` : ''}</strong>{x.filename}
              </a>
            </Show>
            <Show when={!this.supportsBlobs}>
              <strong>{`${i+1}. `}{x.description ? `${x.description} - ` : ''}</strong>{x.filename}
            </Show>
          </td>
          <td>
            <button onClick={this.delete.bind(this, x.id)}>
              <i className="fa fa-trash" aria-hidden="true"></i>
              <span>{i18n.t('application.attachments.upload.files.remove')}</span>
            </button>
          </td>
        </tr>
      )
    })
  }

  limits () {
    const fileMax = env.FileMaximumSize()

    const fileTypes = env.FileTypes()
    fileTypes[fileTypes.length - 1] = `or ${fileTypes[fileTypes.length - 1]}`

    const message = i18n.t('application.attachments.upload.limits')
          .replace('{types}', fileTypes.join(', '))
          .replace('{max_size}', humanReadableFileSize(fileMax))

    return (
      <p>
        <strong>{message}</strong>
      </p>
    )
  }

  render () {
    return (
      <div className="attachments">
        <Field title={i18n.t('application.attachments.method.title')}
               help="application.attachments.help"
               className="attachment-type"
               optional={true}>
          {i18n.m('application.attachments.method.para')}
          <RadioGroup className="attachment-type option-list eapp-extend-labels"
                      onError={this.props.onError}
                      selectedValue={(this.props.AttachmentType || {}).value}>
            <Radio name="attachment-type-upload"
                   label={i18n.m('application.attachments.type.upload')}
                   value="Upload"
                   onUpdate={this.updateAttachmentType}
                   onError={this.props.onError}>
              <div className="attachment-icon upload">
                <Svg src="/img/attach-upload.svg" />
              </div>
            </Radio>
            <Radio name="attachment-type-fax"
                   label={i18n.m('application.attachments.type.fax')}
                   value="Fax"
                   onUpdate={this.updateAttachmentType}
                   onError={this.props.onError}>
              <div className="attachment-icon fax">
                <Svg src="/img/attach-fax.svg" />
              </div>
            </Radio>
            <Radio name="attachment-type-other"
                   label={i18n.m('application.attachments.type.other')}
                   value="Other"
                   onUpdate={this.updateAttachmentType}
                   onError={this.props.onError}>
              <div className="attachment-icon mail">
                <Svg src="/img/attach-mail.svg" />
              </div>
            </Radio>
          </RadioGroup>
        </Field>

        <Show when={(this.props.AttachmentType || {}).value === 'Upload'}>
          <Field title={i18n.t('application.attachments.upload.title')}
                 optional={true}
                 className={this.state.errorMessage ? 'no-margin-bottom upload-area' : 'upload-area'}>
            {i18n.m('application.attachments.upload.para')}
            {this.limits()}
            <form ref="form">
              <input id="file-upload" ref="file" type="file" />
            </form>
            <Text {...this.props.UploadDescription}
                  name="UploadDescription"
                  label={i18n.t('application.attachments.upload.description')}
                  onUpdate={this.updateUploadDescription}
                  onError={this.props.onError}
                  />
            <button onClick={this.uploadFile}>
              {i18n.t('application.attachments.upload.send')}
            </button>
          </Field>

          <Show when={this.state.errorMessage}>
            <div className="field upload-error">
              <div className="table expand">
                <span className="messages error-messages">
                  <div className="message error">
                    <i className="fa fa-exclamation"></i>
                    <h3>{i18n.t('application.attachments.upload.error.title')}</h3>
                    <p>{this.state.errorMessage}</p>
                  </div>
                </span>
              </div>
            </div>
          </Show>

          <Show when={this.state.attachments.length > 0}>
            <Field title={i18n.t('application.attachments.upload.files.title')}
                   optional={true}>
              {i18n.m('application.attachments.upload.files.para')}
              <table>
                <tbody>
                  {this.displayAttachments(this.state.attachments)}
                </tbody>
              </table>
            </Field>
          </Show>
        </Show>

        <Show when={(this.props.AttachmentType || {}).value === 'Fax'}>
          <Field title={i18n.t('application.attachments.fax.title')}
                 optional={true}
                 className="fax-area">
            {i18n.m('application.attachments.fax.para')}
            <button>
              {i18n.t('application.attachments.fax.print')}
            </button>
          </Field>
        </Show>

        <Show when={(this.props.AttachmentType || {}).value === 'Other'}>
          <Field title={i18n.t('application.attachments.other.title')}
                 optional={true}
                 className="other-area">
            {i18n.m('application.attachments.other.para')}
            <Text {...this.props.OtherMethod}
                  name="OtherMethod"
                  label={i18n.t('application.attachments.other.method')}
                  onUpdate={this.updateOtherMethod}
                  onError={this.props.onError}
                  />
          </Field>
        </Show>
      </div>
    )
  }
}

Attachments.defaultProps = {
  errorMessage: '',
  AttachmentType: {},
  UploadDescription: {},
  OtherMethod: {},
  section: 'releases',
  subsection: 'attachments',
  onUpdate: (queue) => {},
  onError: (value, arr) => { return arr }
}

const humanReadableFileSize = (bytes, si = true) => {
  const threshold = si ? 1000 : 1024
  if (Math.abs(bytes) < threshold) {
    return `${bytes} B`
  }

  const units = si
        ? ['kB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB']
        : ['KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB']
  let u = -1;
  do {
    bytes /= threshold
    u++
  } while (Math.abs(bytes) >= threshold && u < units.length - 1)

  return `${bytes.toFixed(1)} ${units[u]}`
}

const blobFromBase64 = (base64, contentType = '', size = 512) => {
  const binary = window.atob(base64)
  const buffer = []
  for (let offset = 0; offset < binary.length; offset += size) {
    let slice = binary.slice(offset, offset + size)
    let numbers =  new Array(slice.length)
    for (let i = 0; i < slice.length; i++) {
      numbers[i] = slice.charCodeAt(i)
    }
    buffer.push(new Uint8Array(numbers))
  }
  return new window.Blob(buffer, { type: contentType })
}
