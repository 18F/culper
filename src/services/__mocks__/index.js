import { getQueryValue } from '../api'

const api = {
  listAttachments: () => new Promise(res => res({ data: [] })),
  samlSLO: () => new Promise(res => res({ data: {} })),
}

export {
  api,
  getQueryValue,
}
