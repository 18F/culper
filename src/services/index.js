import { api, getQueryValue, getCookieValue, deleteCookie } from './api'
export { api, getQueryValue, getCookieValue, deleteCookie }
