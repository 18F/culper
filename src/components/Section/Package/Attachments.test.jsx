import React from 'react'
import MockAdapter from 'axios-mock-adapter'
import { mount } from 'enzyme'
import { api } from '../../../services'
import Attachments from './Attachments'

describe('The attachments component', () => {
  beforeEach(() => {
    const mock = new MockAdapter(api.proxy)
    mock.onGet('/me/attachment').reply(200, {})
  })

  it('displays for uploads', () => {
    const props = {
      AttachmentType: { value: 'Upload' }
    }
    const component = mount(<Attachments {...props} />)
    expect(component.find('.upload-area').length).toBe(1)
  })

  it('displays for uploads with error message', () => {
    const props = {
      AttachmentType: { value: 'Upload' },
      errorMessage: 'This is a test'
    }
    const component = mount(<Attachments {...props} />)
    expect(component.find('.upload-error .message.error p').text()).toBe(props.errorMessage)
  })

  it('displays for fax', () => {
    const props = {
      AttachmentType: { value: 'Fax' }
    }
    const component = mount(<Attachments {...props} />)
    expect(component.find('.fax-area').length).toBe(1)
  })

  it('displays for mail', () => {
    const props = {
      AttachmentType: { value: 'Other' }
    }
    const component = mount(<Attachments {...props} />)
    expect(component.find('.other-area').length).toBe(1)
  })

  it('triggers updates', () => {
    let updates = 0
    const props = {
      AttachmentType: { value: 'Upload' },
      onUpdate: () => { updates++ }
    }
    const component = mount(<Attachments {...props} />)
    expect(component.find('.upload-area').length).toBe(1)
    component.find('input[type="radio"][value="Upload"]').simulate('click')
    expect(updates).toBe(1)
  })
})
