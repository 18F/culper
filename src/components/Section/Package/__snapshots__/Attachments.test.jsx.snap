// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The attachments component renders properly 1`] = `
<div
  className="attachments"
>
  <div
    aria-label="Specify your attachment method"
    className="field   attachment-type"
    data-uuid="field-MOCK-GUID-34"
  >
    <a
      aria-hidden="true"
      className="anchor"
      id="field-MOCK-GUID-34"
      name="field-MOCK-GUID-34"
    />
    <h3
      className="title h3"
    >
      Specify your attachment method
    </h3>
    <span
      className="icon"
    >
      <a
        aria-label="Show help for Specify your attachment method"
        className="toggle h3"
        href="javascript:;"
        onClick={[Function]}
        title="Show help for Specify your attachment method"
      >
        <svg
          alt="Scalable vector graphic"
          focusable="false"
          height="14.57px"
          tabIndex="-1"
          viewBox="0 0 14.57 14.57"
          width="14.57px"
        >
          <g>
            <path
              className="eapp-help-icon"
              d="M13.59,3.63c0.65,1.12,0.98,2.34,0.98,3.66s-0.33,2.54-0.98,3.66c-0.65,1.12-1.54,2-2.65,2.65 s-2.33,0.98-3.66,0.98c-1.32,0-2.54-0.33-3.66-0.98s-2-1.54-2.65-2.65C0.33,9.83,0,8.61,0,7.29s0.33-2.54,0.98-3.66 c0.65-1.12,1.54-2,2.65-2.65S5.96,0,7.29,0c1.32,0,2.54,0.33,3.66,0.98S12.94,2.51,13.59,3.63z M10.93,5.46 c0-0.56-0.18-1.07-0.53-1.55c-0.35-0.47-0.79-0.84-1.31-1.1C8.56,2.56,8.03,2.43,7.48,2.43c-1.54,0-2.71,0.67-3.52,2.02 C3.86,4.6,3.89,4.73,4.03,4.85L5.28,5.8c0.04,0.04,0.1,0.06,0.18,0.06c0.1,0,0.18-0.04,0.24-0.11c0.33-0.43,0.61-0.72,0.82-0.87 C6.73,4.71,7,4.64,7.33,4.64c0.3,0,0.57,0.08,0.81,0.25C8.38,5.05,8.5,5.24,8.5,5.45c0,0.24-0.06,0.43-0.19,0.58 C8.18,6.17,7.97,6.31,7.67,6.45C7.27,6.63,6.9,6.9,6.57,7.27c-0.33,0.37-0.5,0.77-0.5,1.19V8.8c0,0.09,0.03,0.16,0.09,0.22 c0.06,0.06,0.13,0.09,0.22,0.09H8.2c0.09,0,0.16-0.03,0.22-0.09C8.47,8.96,8.5,8.89,8.5,8.8c0-0.12,0.07-0.28,0.2-0.47 c0.14-0.19,0.31-0.35,0.52-0.47c0.2-0.11,0.36-0.21,0.46-0.27s0.25-0.18,0.44-0.33c0.18-0.15,0.32-0.31,0.42-0.46 c0.1-0.15,0.19-0.34,0.27-0.57C10.89,6,10.93,5.74,10.93,5.46z M8.5,11.84v-1.82c0-0.09-0.03-0.16-0.08-0.22 C8.36,9.74,8.29,9.71,8.2,9.71H6.38c-0.09,0-0.16,0.03-0.22,0.08C6.1,9.86,6.07,9.93,6.07,10.02v1.82c0,0.09,0.03,0.16,0.09,0.22 c0.06,0.06,0.13,0.08,0.22,0.08H8.2c0.09,0,0.16-0.03,0.22-0.08C8.47,12,8.5,11.93,8.5,11.84z"
            />
          </g>
        </svg>
      </a>
    </span>
    <div
      className="table expand"
    >
      <span
        aria-live="polite"
        className="messages help-messages"
      />
    </div>
    <div
      className="table"
    >
      <span
        className="content"
      >
        <span
          className="component"
        >
          <div>
            <p>
              Choose the method you will use to provide attachments for your Investigation Request.
            </p>
          </div>
          <div
            className="blocks attachment-type option-list eapp-extend-labels"
          >
            <div
              className="block extended"
            >
              <label
                className="checked"
                htmlFor="attachment-type-upload-MOCK-GUID-36"
              >
                <input
                  aria-label="[object Object] for null"
                  checked={true}
                  className="usa-input-success selected"
                  disabled={false}
                  id="attachment-type-upload-MOCK-GUID-36"
                  name="attachment-type-upload-MOCK-GUID-36"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  type="radio"
                  value="Upload"
                />
                <div
                  className="attachment-icon upload"
                >
                  <img
                    alt="Scalable vector graphic"
                    className="svg"
                    src="/img/attach-upload.svg"
                  />
                </div>
                <span>
                  <div>
                    <p>
                      Upload file
                    </p>
                  </div>
                </span>
              </label>
            </div>
            <div
              className="block extended"
            >
              <label
                className=""
                htmlFor="attachment-type-fax-MOCK-GUID-37"
              >
                <input
                  aria-label="[object Object] for null"
                  checked={false}
                  className="usa-input-success"
                  disabled={false}
                  id="attachment-type-fax-MOCK-GUID-37"
                  name="attachment-type-fax-MOCK-GUID-37"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  type="radio"
                  value="Fax"
                />
                <div
                  className="attachment-icon fax"
                >
                  <img
                    alt="Scalable vector graphic"
                    className="svg"
                    src="/img/attach-fax.svg"
                  />
                </div>
                <span>
                  <div>
                    <p>
                      Direct fax
                    </p>
                  </div>
                </span>
              </label>
            </div>
            <div
              className="block extended"
            >
              <label
                className=""
                htmlFor="attachment-type-other-MOCK-GUID-38"
              >
                <input
                  aria-label="[object Object] for null"
                  checked={false}
                  className="usa-input-success"
                  disabled={false}
                  id="attachment-type-other-MOCK-GUID-38"
                  name="attachment-type-other-MOCK-GUID-38"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  type="radio"
                  value="Other"
                />
                <div
                  className="attachment-icon mail"
                >
                  <img
                    alt="Scalable vector graphic"
                    className="svg"
                    src="/img/attach-mail.svg"
                  />
                </div>
                <span>
                  <div>
                    <p>
                      Other
                    </p>
                  </div>
                </span>
              </label>
            </div>
          </div>
        </span>
      </span>
    </div>
    <div
      className="table expand"
    >
      <span
        aria-live="polite"
        className="messages error-messages"
        role="alert"
      />
    </div>
  </div>
  <span>
    <div
      aria-label="Upload file to e-QIP directly"
      className="field   upload-area"
      data-uuid="field-MOCK-GUID-39"
    >
      <a
        aria-hidden="true"
        className="anchor"
        id="field-MOCK-GUID-39"
        name="field-MOCK-GUID-39"
      />
      <h3
        className="title h3"
      >
        Upload file to e-QIP directly
      </h3>
      <span
        className="icon"
      />
      <div
        className="table expand"
      >
        <span
          aria-live="polite"
          className="messages help-messages"
        />
      </div>
      <div
        className="table"
      >
        <span
          className="content"
        >
          <span
            className="component"
          >
            <div>
              <p>
                Certification, General Release and Medical Release forms must be attached separately as single page documents. Items such as a resume or OF 612 can be uploaded as multi-page documents.
              </p>
            </div>
            <p>
              <strong>
                TIFF, PNG, or PDF files only. Maximum file size allowed is 5.0 MB.
              </strong>
            </p>
            <form>
              <input
                id="file-upload"
                type="file"
              />
            </form>
            <div
              className=""
            >
              <label
                className=""
                htmlFor="UploadDescription-MOCK-GUID-40"
              >
                Description
              </label>
              <input
                aria-describedby="UploadDescription-error"
                aria-label="Description"
                autoCapitalize={true}
                autoComplete={true}
                autoCorrect={true}
                className=""
                id="UploadDescription-MOCK-GUID-40"
                maxLength={255}
                name="UploadDescription"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                pattern=".*"
                spellCheck={true}
                type="text"
                value=""
              />
            </div>
            <button
              onClick={[Function]}
            >
              Upload attachment
            </button>
          </span>
        </span>
      </div>
      <div
        className="table expand"
      >
        <span
          aria-live="polite"
          className="messages error-messages"
          role="alert"
        />
      </div>
    </div>
  </span>
</div>
`;
